presto-main/src/test/java/com/facebook/presto/operator/exchange/TestLocalExchange.java

```diff
@@ -235,13 +238,17 @@ public void testPartition()
         LocalExchangeSource sourceB = exchange.getSource(1);
         assertSource(sourceB, 0);
 
-        sink.addPage(createPage(0));
```

Comment:
Can you add a test where you add only a partial page, and where you have values that are in different partitions?